### PR TITLE
Feat(eos_designs): Schema for management

### PR DESCRIPTION
**Pull request by carlbuchmann**
# EOS Designs Schema - management

Schema for the following keys:

## Contributor Checklist

- [x] Review documentation for accuracy and completeness and compare with old documentation.
- [x] Ensure `convert_types` is leveraged were applicable to auto-convert, accepting `str` in `int` and `int` in `str`.
- [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
- [x] run `pre-commit run schemas --all`
- [x] Verify that CI pass.

## Reviewer Checklist

**Reviewer 1:**

- [ ] Review documentation for accuracy and completeness and compare with old documentation.
- [ ] Ensure `convert_types` is leveraged were applicable to auto-convert, accepting `str` in `int` and `int` in `str`.
- [ ] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
- [ ] Verify that CI pass.

**Reviewer 2:**

- [ ] Review documentation for accuracy and completeness and compare with old documentation.
- [ ] Ensure `convert_types` is leveraged were applicable to auto-convert, accepting `str` in `int` and `int` in `str`.
- [ ] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
- [ ] Verify that CI pass.

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/name_servers.schema.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  name_servers:
+    type: list
+    description: Only eos_designs name_servers variables.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/name_servers.schema.yml]: Not the best description.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/name_server.schema.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  name_server:
+    type: dict
+    description: |
+      Only name_server from eos_ci_config_gen.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/name_server.schema.yml]: I don't think we want eos_cli_config_gen keys here (yet)

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_vrf_routing.schema.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  mgmt_vrf_routing:
+    type: bool
+    default: False
+    description: Management interface configuration.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_vrf_routing.schema.yml]: misleading description. Better to leave it out if we don't have a specific (good) description. Same elsewhere for management vars.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/inband_management_subnet.schema.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  inband_management_subnet:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/inband_management_subnet.schema.yml]: I don't think we support this at the root level.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/inband_management_vlan.schema.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  inband_management_vlan:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/inband_management_vlan.schema.yml]: I don't think we support this at the root level.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/event_handlers.schema.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+keys:
+  event_handlers:
+    $ref: "eos_cli_config_gen#/keys/event_handlers"
+    type: list
+    documentation_options:
+      filename: Input Variables
+    description: |
+      Gives the ability to monitor and react to Syslog messages.
+      Event Handlers provide a powerful and flexible tool that can be used to apply self-healing actions,
+      customize the system behavior, and implement workarounds to problems discovered in the field.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/event_handlers.schema.yml]: Could this go directly in `eos_cli_config_gen` and be retrived with the `$refs` ?

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_gateway.schema.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  mgmt_gateway:
+    type: str
+    description: OOB Management interface gateway in IPv4 format.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_gateway.schema.yml]: Let's add that by default it is used to configure a default route except if overriden by <mgmt_destination_networks>

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        type: bool
+        default: False
+        description: Set SNMP location. Formatted as {{ fabric_name }} {{ dc_name }} {{ pod_name }} {{ switch_rack }} {{ inventory_hostname }}.
+      compute_local_engineid:
+        type: bool
+        default: False
+        description: |
+          Generate a local engineId for SNMP by hashing via SHA1, the string
+          generated via the concatenation of the hostname plus the management IP.
+          {{ inventory_hostname }} + {{ switch.mgmt_ip }}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml]: Fixing to latest description (from current doc) - not sure why it was changed?
```suggestion
          Generate a local engineId for SNMP using the 'compute_local_engineid_source' method.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+          - system_mac
+      compute_v3_user_localized_key:
+        type: bool
+        default: False
+        description: |
+          Requires compute_local_engineid to be `true`. If enabled, the SNMPv3
+          passphrases for auth and priv are transformed using RFC 2574,
+          matching the value they would take in EOS CLI. The algorithm requires
+          a local engineId, which is unknown to AVD, hence the necessity to generate
+          one beforehand.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml]:
```suggestion
          Requires compute_local_engineid to be `true`. 
          If enabled, the SNMPv3 passphrases for auth and priv are transformed using RFC 2574, matching the value they would take in EOS CLI.
          The algorithm requires a local engineId, which is unknown to AVD, hence the necessity to generate one beforehand.
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+      users:
+        type: list
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Username
+            group:
+              type: str
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/snmp_settings.schema.yml:52]: Here I would add in the description that for now the groups are not automatically generated and needs to be added using `structured_config` (I know my bad)

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/terminattr_ingestgrpcurl_port.schema.yml [hunk 11]
```diff
@@ -11,4 +11,6 @@ keys:
     convert_types:
     - str
     default: 9910
-    description: Port number for Terminattr ingest GRPC
+    description: |-
+      Port number for Terminattr connection to an on-premise CloudVision cluster.
+      The port number is always 443 when using CloudVision as a Service.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/terminattr_ingestgrpcurl_port.schema.yml]: Does this mean AVD handles making it `443` or should the user configure it?

**Review by gmuloc — COMMENTED** [review]
seems like we are missing `system_mac_address` in the schema?
https://avd.sh/en/stable/roles/eos_designs/doc/management-settings.html

**Review by gmuloc — COMMENTED** [review]
seems like we are missing `system_mac_address` in the schema?
https://avd.sh/en/stable/roles/eos_designs/doc/management-settings.html

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/event_handlers.schema.yml (hunk 6 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/event_handlers.schema.yml]: done

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_gateway.schema.yml (hunk 7 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/mgmt_gateway.schema.yml]: done

**Review by ClausHolbechArista — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Comment by gmuloc** [comment]
CI failure can be ignored for mkdocs and has been fixed in devel already

**Pull request merged**